doc/index.md

```diff
@@ -1,10 +1,103 @@
 Welcome to svgdigitizer's documentation!
 ========================================
 
+<!--
 ```{todo}
 * what is svgdigitizer and what is our aim.
 * then refer to installation, cli, api and cv.
 ```
+-->
+
+Svgdigitizer allows recovering data from a curve, plotted in a 2D coordinate 
+system with defined axes.
+Such plots are often found in scientific publications, where
+in many cases, especially for old pulications, the underlying data 
+is not accessible anymore. 
+In some cases, the axis of the plot can be a bit skewed, caused from scanning 
+printed documents. An extreme case for such a plot is depicted in the following figure.
+
+![files/images/example_plot_p0.png](files/images/example_plot_p0.png) 
+
+In order to recover the underlying data, first the plot is imported in a 
+vector graphics program, such as [Inkscape](https://inkscape.org/).
+The curve is traced with *regular bezier path* and two points are marked 
+with a group of textlabel and line on each axis to define the coordinate system.
+The curve should be assigned a text label. Additonal labels describing the data 
+can be provided anywhere in the SVG file. The resulting file looks as follows:
+
+![files/images/example_plot_p0_demo.png](files/images/example_plot_p0_demo.png) 
+
+## [Command line interface](cli.md)
+This file can be digitized from the [command line interface](cli.md), which creates a CVS file of the x and y data. 
+The resolution can be specified by `--sampling-interval`.
+
+```sh
+svgdigitizer digitize example_plot_p0_demo.svg --sampling-interval 0.01
+```
+
+## [API](api.md)
+With the [API](api.md), the file can also be used to create an actual [SVGplot class](api/svgplot.md).
+
+```python
+>>> from svgdigitizer.svg import SVG
+>>> from svgdigitizer.svgplot import SVGPlot
+
+>>> plot = SVGPlot(SVG(open('./svgdigitizer/doc/files/others/example_plot_p0_demo.svg', 'r')), sampling_interval=0.01)
```

Comment:
```suggestion
>>> plot = SVGPlot(SVG(open('./svgdigitizer/doc/files/others/example_plot_p0_demo.svg', 'r')), sampling_interval=0.01, algorithm='mark-aligned')
```

The example plot below looks wrong to me at the moment. The graph is tilted upwards a bit as it has been in the original drawing. But it should be flat because the x-axis is not horizontal.

Reply:
fixed the plot. weird enough 'r' worked